### PR TITLE
Develop3

### PR DESCRIPTION
**Pull request by manhdd-1633**
Em đẩy pull này hơi nhiều vì hôm nay demo ạ..

### CODE REVIEW

File: app/Http/Controllers/Admin/DiagramController.php [hunk 1]
```diff
@@ -94,27 +104,49 @@ public function generateDiagram(Request $request, $idWorkspace)
         }
         $locations = $workspace->locations;
         $colorLocation = [];
+
         foreach ($locations as $key => $location) {
             foreach ($location->seats as $id => $seat) {
+                $user_id = $this->userRepository->get();
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/DiagramController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/DiagramController.php</br>**Line**: 110</br>Variable "user_id" is not in valid camel caps format

File: app/Http/Controllers/Admin/DiagramController.php [hunk 2]
```diff
@@ -94,27 +104,49 @@ public function generateDiagram(Request $request, $idWorkspace)
...
+                $list_user_id = unserialize($seat->user_id);
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/DiagramController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/DiagramController.php</br>**Line**: 111</br>Variable "list_user_id" is not in valid camel caps format

File: app/Http/Controllers/Admin/DiagramController.php [hunk 3]
```diff
@@ -94,27 +104,49 @@ public function generateDiagram(Request $request, $idWorkspace)
...
                 $colorLocation[$key][$id]['seat_id'] = $seat->id;
-                if ($seat->user != null) {
-                    $colorLocation[$key][$id]['user_name'] = $seat->user->name;
-                    $colorLocation[$key][$id]['avatar'] = $seat->user->avatar;
-                    $colorLocation[$key][$id]['user_id'] =  $seat->user->id;
+                if ($seat->user_id != null) {
+                    $checkName = [];
+                    $checkAvatar = [];
+                    $checkUserId = [];
+                    foreach ($user_id as $value) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/DiagramController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/DiagramController.php</br>**Line**: 118</br>Variable "user_id" is not in valid camel caps format

File: app/Http/Controllers/Admin/DiagramController.php [hunk 4]
```diff
@@ -94,27 +104,49 @@ public function generateDiagram(Request $request, $idWorkspace)
...
+                        if (in_array($value->id, $list_user_id)) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/DiagramController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/DiagramController.php</br>**Line**: 119</br>Variable "list_user_id" is not in valid camel caps format

File: app/Http/Requests/DesignDiagramRequests.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'name' => 'required|unique:design_diagram,name',
```
Comment (framgiaciv3) [inline app/Http/Requests/DesignDiagramRequests.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Http/Requests/DesignDiagramRequests.php</br>**Line**: 29</br>Each line in an array declaration must end in a comma

File: app/Http/Requests/DesignDiagramRequests.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+            'diagram' => 'required|nullable',
+            'content' => 'required'
+        ];
+    }
+
+    public function messages()
```
Comment (framgiaciv3) [inline app/Http/Requests/DesignDiagramRequests.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.LanguageConstructSpacing.Incorrect</br>**Severity**: error</br>**File**: app/Http/Requests/DesignDiagramRequests.php</br>**Line**: 35</br>Language constructs must be followed by a single space; expected "return [" but found "return["

File: app/Http/Requests/DesignDiagramRequests.php [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    {
```
Comment (framgiaciv3) [inline app/Http/Requests/DesignDiagramRequests.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoSpaceAfterComma</br>**Severity**: error</br>**File**: app/Http/Requests/DesignDiagramRequests.php</br>**Line**: 36</br>Expected 1 space between "'name.required'" and double arrow; 0 found

File: app/Http/Requests/DesignDiagramRequests.php [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        return[
+            'name.required'=> __('Required'),
+            'name.unique' => __('Exists'),
```
Comment (framgiaciv3) [inline app/Http/Requests/DesignDiagramRequests.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Http/Requests/DesignDiagramRequests.php</br>**Line**: 39</br>Each line in an array declaration must end in a comma

File: app/Http/Requests/DesignDiagramRequests.php [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    public function rules()
+    {
+        return [
+            'name' => 'required|unique:design_diagram,name',
+            'diagram' => 'required|nullable',
+            'content' => 'required',
+        ];
+    }
+
+    public function messages()
```
Comment (framgiaciv3) [inline app/Http/Requests/DesignDiagramRequests.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.LanguageConstructSpacing.Incorrect</br>**Severity**: error</br>**File**: app/Http/Requests/DesignDiagramRequests.php</br>**Line**: 35</br>Language constructs must be followed by a single space; expected "return [" but found "return["

File: app/Http/Requests/DesignDiagramRequests.php [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+    {
```
Comment (framgiaciv3) [inline app/Http/Requests/DesignDiagramRequests.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoSpaceAfterComma</br>**Severity**: error</br>**File**: app/Http/Requests/DesignDiagramRequests.php</br>**Line**: 36</br>Expected 1 space between "'name.required'" and double arrow; 0 found

Comment (framgiaciv3) [inline app/Http/Requests/DesignDiagramRequests.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoSpaceAfterComma</br>**Severity**: error</br>**File**: app/Http/Requests/DesignDiagramRequests.php</br>**Line**: 36</br>Expected 1 space between "'name.required'" and double arrow; 0 found

Comment (framgiaciv3) [inline app/Http/Requests/DesignDiagramRequests.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoSpaceAfterComma</br>**Severity**: error</br>**File**: app/Http/Requests/DesignDiagramRequests.php</br>**Line**: 36</br>Expected 1 space between "'name.required'" and double arrow; 0 found

File: app/Http/Requests/DesignDiagramRequests.php [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+        return [
+            'name.required'=> __('Required'),
+            'name.unique' => __('Exists'),
```
Comment (framgiaciv3) [inline app/Http/Requests/DesignDiagramRequests.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Http/Requests/DesignDiagramRequests.php</br>**Line**: 39</br>Each line in an array declaration must end in a comma

File: app/Http/Controllers/Admin/DiagramController.php [hunk 12]
```diff
@@ -253,6 +293,82 @@ public function editInfoLocation(Request $request)
...
+        try {
+            if ($request->diagram) {
+                $request->diagram->store(config('site.diagram.image'));
+                $data['diagram'] = $request->diagram->hashName();
+            }
+            $this->designDiagramRepository->create($data);
+            DB::commit();
+            Alert::success(trans('Add success'), trans('Successfully!!!'));
+        } catch (Exception $e) {
+            DB::rollback();
```
Comment (BunnyPi04) [inline app/Http/Controllers/Admin/DiagramController.php:315]: vừa try catch vừa transaction à :v

File: resources/css/image_map.css [hunk 13]
```diff
@@ -181,4 +187,24 @@ footer .container {
...
+#image-mapper-table-1 {
+    width: 25px
+}
+#image-mapper-table-2 {
+   width: 65px
+}
+#add-new-area {
+    text-align: right
+}
+#show-me-the-code {
```
Comment (BunnyPi04) [inline resources/css/image_map.css:204]: `show-me-the-code`
(facepalm)

File: resources/js/generate.js [hunk 14]
```diff
@@ -111,35 +135,76 @@ $(document).ready(function () {
...
+                url: url+id,
                 success: function (data) {
-                    var url ='../storage/user/'+ data.avatar;
-                        if (url) {
-                            $('#modal-info-user').find('#imagePreview').css('background-image', 'url(' + url + ')');
-                        }
-                        $('#modal-info-user').find('#list-name option[value = '+data.id+']').attr('selected','selected');
-                        $('#modal-info-user').find('#list-language option[value = '+data.program_id+']').attr('selected','selected');
-                        $('#modal-info-user').find('#list-position option[value = '+data.position_id+']').attr('selected','selected');
+                    var avatar = '../storage/user/'+data.avatar;
```
Comment (BunnyPi04) [inline resources/js/generate.js]: convention

File: resources/js/generate.js [hunk 15]
```diff
@@ -81,15 +110,10 @@ $(document).ready(function () {
...
-        if (avatar) {
-            $(this).append('<div id="hello"><span><small><i class="fa fa-circle" style="color: green"></i></small> ' + checkID + ' <br> <img src="storage/user/'+avatar+'" alt="" style = "width: 40px; "></span></div>');
-        } else {
-            $(this).append('<div id="hello"><span><small><i class="fa fa-circle" style="color: green"></i></small> ' + checkID + ' <br> </span></div>');
-        }
-
+    $('.all_seat .seat button').hover(function() {
+        var checkID = $(this).attr('id');
+        var checkAvatar = $(this).attr('info-avatar');
+        $(this).append('<div id="hello"><span><small><i class="fa fa-circle" style="color: green; padding-right: 10px"></i></small> ' + checkID + ' <br> <br> <img src="storage/user/' + checkAvatar + '" alt="" style = "width: 40px; "> </span></div>');
```
Comment (BunnyPi04) [inline resources/js/generate.js]: đẩy style vào css đi

File: resources/views/test/workspace/image_map.blade.php [hunk 16]
```diff
@@ -5,82 +5,88 @@
...
+                                </div>
                             </div>
                         </div>
                     </div>
                 </div>
                 <table class="table" id="image-mapper-table">
                     <thead>
                         <tr>
-                            <th>Active</th>
+                            <th>Active </th>
```
Comment (BunnyPi04) [inline resources/views/test/workspace/image_map.blade.php]: i18n

File: resources/views/test/workspace/index.blade.php [hunk 17]
```diff
@@ -74,4 +77,65 @@
             </div>
         </div>
     </div>
+    <div class="modal fade" id="m_modal_4" tabindex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true" style="display: none;">
+        <div class="modal-dialog modal-lg" role="document">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <h5 class="modal-title" id="exampleModalLabel">New message</h5>
```
Comment (BunnyPi04) [inline resources/views/test/workspace/index.blade.php]: i18n

File: app/Http/Controllers/Admin/DiagramController.php (hunk 12 above)
Comment (manhdd-1633) [reply app/Http/Controllers/Admin/DiagramController.php:315]: Có sao đâu ạ.. bản chất như vậy còn gì ạ.. try & catch để xuất ra lỗi chứ ạ

File: resources/css/image_map.css (hunk 13 above)
Comment (manhdd-1633) [reply resources/css/image_map.css:204]: tên id như vậy có vẫn đề gì ạ

File: resources/js/generate.js (hunk 14 above)
Comment (manhdd-1633) [reply resources/js/generate.js]: ready

File: resources/js/generate.js [hunk 18]
```diff
@@ -111,35 +135,76 @@ $(document).ready(function () {
...
-        var user_id = $(this).attr('user_id');
-        $('#locations').val(user_id);
-        var avatar = $(this).attr('avatar');
-        if (avatar != '') {
-            var id = $(this).attr('id');
+        var seat_id = $(this).attr('seat_id');
+        $('#locations').val(seat_id);
+    });
+
+    $(document).on('click',  '.seat button', function() {
```
Comment (BunnyPi04) [inline resources/js/generate.js]: `'click',  '.seat button'`
dư 1 space này

File: resources/js/generate.js [hunk 19]
```diff
@@ -2,28 +2,57 @@ $(document).ready(function () {
...
-            $('#' + id + '').html(name_user[name_user.length -1]);
+
+            var x = '';
+            if (Array.isArray(user_name) && Array.isArray(avatar)) {
+                var result =  Object.assign.apply({}, user_name.map( (v, i) => ( {[v]: avatar[i]} ) ) );
+                var count_user = 0;
+                for (var k in result) {
+                    if (result.hasOwnProperty(k)) {
+                        if (user_name.length <= 1) {
+                            x += '<button type="button" class="btn m-btn--pill  btn-outline-success button-hover"  data-toggle="modal" href="#modal-info-user-1" data-id="'+user_id[count_user]+'" position = "' + position + '"program="' + program + '" id="' + k + '" info-avatar="'+ result[k] +'">' + status + k + ' </button><br><a class="btn m-btn--pill m-btn--air btn-secondary" data-toggle="modal" data-target="#modal-info-user" data-placement="left" data-original-title="Add Workspace"> <i class="flaticon-add"></i></a>';
```
Comment (BunnyPi04) [inline resources/js/generate.js]: `button-hover"  data-toggle`
dư 1 space này

File: resources/js/generate.js [hunk 20]
```diff
@@ -2,28 +2,57 @@ $(document).ready(function () {
...
+            var user_name = paintLocation[i][j]['user_name'];
             var avatar = paintLocation[i][j]['avatar'];
             var color = paintLocation[i][j]['color'];
             var user_id = paintLocation[i][j]['user_id'];
+            var program = paintLocation[i][j]['program'];
+            var position = paintLocation[i][j]['position'];
+            var status = '<small><i class="fa fa-circle" style="color: green; padding-right: 10px"></i></small>';
             $('#' + id + '').attr('seat_id', paintLocation[i][j]['seat_id']);
-            $('#' + id + '').css('background-color', color);
+            $('#' + id + '').css({
```
Comment (BunnyPi04) [inline resources/js/generate.js:22]: `$('#' + id + '')`
nối chuỗi với rỗng thì nối làm gì nhỉ?

File: resources/js/generate.js [hunk 21]
```diff
@@ -111,35 +135,76 @@ $(document).ready(function () {
...
+
+    $(document).on('click',  '.seat button', function() {
+        var seat_id = $(this).closest('span').attr('seat_id');
+        var user_id = $(this).data('id');
+        $('#locations-1').val(seat_id);
+        $('#user-id-current').val(user_id);
+
+    });
+
+    $( '#list-name' ).change(function () {
```
Comment (BunnyPi04) [inline resources/js/generate.js:150]: convention

File: resources/js/generate.js (hunk 14 above)
Comment (BunnyPi04) [reply resources/js/generate.js]: sửa convention tương tự nữa

File: resources/js/generate.js [hunk 22]
```diff
@@ -111,35 +135,76 @@ $(document).ready(function () {
...
+                    'X-CSRF-TOKEN': $('meta[name="csrf-token"]').attr('content')
                 }
             });
-        };
+            $.ajax({
+                type: 'post',
+                url: url+id,
+                success: function (data) {
+                    if (data) {
+                        var avatar = '../storage/user/' + data.avatar;
```
Comment (BunnyPi04) [inline resources/js/generate.js]: base đưa vào config nhé

File: resources/js/generate.js [hunk 23]
```diff
@@ -111,35 +135,76 @@ $(document).ready(function () {
...
+                        $('#modal-info-user-1').find('#imagePreview').css('background-image','url(' + avatar + ')');
```
Comment (BunnyPi04) [inline resources/js/generate.js]: convention

File: resources/js/generate.js (hunk 20 above)
Comment (manhdd-1633) [reply resources/js/generate.js:22]: trời chuỗi rỗng đâu bạn.. xem lại hộ mình với, nó sẽ là thế này = > $('#id')

File: app/Http/Controllers/Admin/DiagramController.php [hunk 24]
```diff
@@ -43,6 +52,12 @@ public function typeWorkspaceInformation()
 
     public function saveWorkspace(Request $request)
     {
+        // $data = $request->all();
+        // if ($request->image) {
+        //     $request->image->store(config('site.workspace.image'));
+        //     $data['image'] = $request->image->hashName();
+        // }
+        // $workspace = $this->workspace->create($data);
```
Comment (BunnyPi04) [inline app/Http/Controllers/Admin/DiagramController.php]: ko dùng xóa đi

File: resources/js/config.js [hunk 25]
```diff
@@ -0,0 +1,4 @@
+var urlAvatar = '../storage/user/';
+
+
+
```
Comment (BunnyPi04) [inline resources/js/config.js]: xóa các dòng trắng dư

File: resources/js/generate.js [hunk 26]
```diff
@@ -111,35 +134,75 @@ $(document).ready(function () {
...
                 success: function (data) {
-                    var url ='../storage/user/'+ data.avatar;
-                        if (url) {
-                            $('#modal-info-user').find('#imagePreview').css('background-image', 'url(' + url + ')');
-                        }
-                        $('#modal-info-user').find('#list-name option[value = '+data.id+']').attr('selected','selected');
-                        $('#modal-info-user').find('#list-language option[value = '+data.program_id+']').attr('selected','selected');
-                        $('#modal-info-user').find('#list-position option[value = '+data.position_id+']').attr('selected','selected');
+                    var avatar = urlAvatar + data.avatar;
+                    $('#modal-info-user').find('#imagePreview').css('background-image','url(' + avatar + ')');
```
Comment (BunnyPi04) [inline resources/js/generate.js]: convention

File: resources/js/generate.js [hunk 27]
```diff
@@ -111,35 +134,75 @@ $(document).ready(function () {
...
-                    }
+                    'X-CSRF-TOKEN': $('meta[name="csrf-token"]').attr('content')
+                }
             });
             $.ajax({
-                type: 'POST',
-                url: url,
-                data: {'user_id' : user_id},
+                type: 'get',
+                url: url+id,
```
Comment (BunnyPi04) [inline resources/js/generate.js]: convention

File: resources/js/generate.js [hunk 28]
```diff
@@ -111,35 +134,75 @@ $(document).ready(function () {
...
+            var url = 'workspace/avatar-info/';
+            var id = $(this).val();
+            $.ajaxSetup({
+                headers: {
+                    'X-CSRF-TOKEN': $('meta[name="csrf-token"]').attr('content')
+                }
+            });
+            $.ajax({
+                type: 'post',
+                url: url+id,
```
Comment (BunnyPi04) [inline resources/js/generate.js]: convention

File: resources/views/test/workspace/image_map.blade.php [hunk 29]
```diff
@@ -117,17 +127,20 @@
...
             <div class="modal-content" id="modal-code-dialog">
                 <div class="modal-header">
-                    <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
-                    <h4 class="modal-title" id="modal-code-label">Generated Image Map Output</h4></div>
+                    <h4 class="modal-title" id="modal-code-label">{{ __('Generated Image Map Output') }}</h4>
+                    {!! Form::button('&times;', ['class' => 'close', 'data-dismiss' => 'modal', 'aria-hidden' => 'true']) !!}
+                </div>
                 <div class="modal-body">
-                    <textarea class="form-control input-sm" readonly="readonly" id="modal-code-result" rows="10"></textarea>
+                    {!! Form::textarea('content' , null , ['class' => 'form-control input-sm', 'readonly'=>'readonly', 'id' => 'modal-code-result', 'row' => '10' ]) !!}
```
Comment (BunnyPi04) [inline resources/views/test/workspace/image_map.blade.php]: convention

File: resources/views/test/workspace/index.blade.php [hunk 30]
```diff
@@ -74,4 +77,64 @@
...
+                                        </div>
+                                        <div class="form-group m-form__group row">
+                                            {!! Form::label(__('Image Office')) !!}
+                                            <div class="custom-file">
+                                                {!! Form::file('image', ['class' => 'custom-file-input', 'id' => 'input-display']) !!}
+                                                {!! Form::label(__('Choose file'), null, ['class' => 'custom-file-label']) !!}
+                                                {!! $errors->first('image', '<p class="text-danger">:message</p>') !!}
+                                            </div>
+                                          {{--   {!! Html::image(' ', null, ['id' => 'image-display', 'class' => 'w-50']) !!} --}}
+                                            <img src="" alt="" id="image-display">
```
Comment (BunnyPi04) [inline resources/views/test/workspace/index.blade.php:119]: ko dùng bỏ đi

File: resources/views/test/workspace/modal_info_user_1.blade.php [hunk 31]
```diff
@@ -0,0 +1,61 @@
...
+                                    <div class="form-group m-form__group row">
+                                        {!! htmlspecialchars_decode( Form::label( 'language', __('Language').'<span class="required">*</span>', [ 'class' => 'control-label col-md-4 col-sm-4 col-xs-12']  ) )  !!}
+                                        <div class="col-8">
+                                            {!! Form::select('language',$listProgram, null, ['class' => 'form-control m-input list-program', 'id' => 'list-language'])  !!}
+                                        </div>
+                                    </div>
+                                    <div class="form-group m-form__group row">
+                                        {!! htmlspecialchars_decode( Form::label( 'position', __('Position').'<span class="required">*</span>', [ 'class' => 'control-label col-md-4 col-sm-4 col-xs-12']  ) )  !!}
+                                        <div class="col-8">
+                                            {!! Form::select('position',$listPosition, null, ['class' => 'form-control m-input list-position', 'id' => 'list-position'])  !!}
```
Comment (BunnyPi04) [inline resources/views/test/workspace/modal_info_user_1.blade.php]: convention
sửa các chỗ tương tự

**Pull request merged**